api/src/main/java/org/apache/iceberg/view/ReplaceViewVersion.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import org.apache.iceberg.PendingUpdate;
+
+/**
+ * API for replacing a view's version.
+ *
+ * <p>Apply returns the updated view version for validation.
+ *
+ * <p>When committing, these changes will be applied to the current view metadata. Commit conflicts
+ * will be resolved by applying the pending changes to the new view metadata.
+ */
+public interface ReplaceViewVersion
+    extends PendingUpdate<ViewVersion>, VersionBuilder<ReplaceViewVersion> {
+
+  /**
+   * Remove the representation for the given dialect from the view's version.
+   *
+   * @param dialect The dialect for which the view representation should be removed.
+   * @return this for method chaining
+   */
+  ReplaceViewVersion remove(String dialect);
```

Comment:
I don't think there's a dialect to remove. Everything must be added specifically for this API right?

Reply:
The `.remove()` is mainly to cover this particular use case:
```
View view = catalog.buildView(identifier)
  .withSchema(schema)
  .withQuery("spark", "sql")
  .withQuery("trino", "sql")
  .createOrReplace()

// updates the SQL for trino and removes the representation for spark
view.replaceViewVersion()
  .withQuery("trino", "updated sql")
  .remove("spark")
  .commit()
```

Reply:
@nastra, I don't think that we want an API that carries over _anything_ from the last version. That's dangerous, so we should remove this method.

Reply:
ok I've removed that method.